drivers/wifi/eswifi/eswifi_socket_offload.c

```diff
@@ -575,8 +575,102 @@ NET_SOCKET_REGISTER(eswifi, AF_UNSPEC, eswifi_socket_is_supported,
 		    eswifi_socket_create);
 #endif
 
+static int eswifi_off_getaddrinfo(const char *node, const char *service,
+				  const struct zsock_addrinfo *hints,
+				  struct zsock_addrinfo **res)
+{
+	struct zsock_addrinfo *ai;
+	struct sockaddr *ai_addr;
+	unsigned long port = 0;
+	char *rsp;
+	int err;
+
+	if (!node) {
+		return DNS_EAI_NONAME;
+	}
+
+	if (service) {
+		port = strtol(service, NULL, 10);
+		if (port < 1 || port > USHRT_MAX) {
+			return DNS_EAI_SERVICE;
+		}
+	}
+
+	if (!res) {
+		return DNS_EAI_NONAME;
+	}
+
+	if (hints && hints->ai_family != AF_INET) {
+		return DNS_EAI_FAIL;
+	}
+
+	eswifi_lock(eswifi);
+
+	/* DNS lookup */
+	snprintk(eswifi->buf, sizeof(eswifi->buf), "D0=%s\r", node);
+	err = eswifi_at_cmd_rsp(eswifi, eswifi->buf, &rsp);
+	if (err < 0) {
+		err = DNS_EAI_FAIL;
+		goto done_unlock;
+	}
+
+	/* Allocate out res (addrinfo) struct.	Just one. */
+	*res = calloc(1, sizeof(struct zsock_addrinfo));
+	ai = *res;
+	if (!ai) {
+		err = EAI_MEMORY;
+		goto done_unlock;
+	}
+
+	/* Now, alloc the embedded sockaddr struct: */
+	ai_addr = calloc(1, sizeof(struct sockaddr));
```

Comment:
You should never instantiate `struct sockaddr` type variables as they might not have enough space for all the IPv[4|6] fields. So depending on the type of the address, you should instead create `struct sockaddr_in` or `struct sockaddr_in6` variable and then do a cast of `struct sockaddr` to correct size struct.

Reply:
Indeed, fixed now.